### DISCUSSION TITLE
Await error

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1231] [Q&A]
I wrote a sample code.. I am not done yet but I am hit with a problem 
For the line ```await sio.connect(URL)```
It says ```"await" allowed only within async function)```
It is the version 4.6.1 which i installed this way ```pip install python-socketio==4.6.1```

**Comment by miguelgrinberg** [comment]
I suggest you learn how to work with asyncio before using this library. The problem that you have is unrelated to Socket.IO, you just need to learn how asyncio works first.

**Reply by Notadumdum** [reply]
I still dont get why i am seeing this error.. I looked at a sample code and it's written the way I wrote mine to qhich you replied the person because he had a problem not relating to ```"await" allowed only within async function```
